src/app/shared/components/tree/tree.component.spec.ts

```diff
@@ -0,0 +1,119 @@
+          nodeName: 'Part paid',
+          id: '104'
+        }
+      ]
+    }
+  ] as TreeData[];
+}
+
+describe('TreeComponent', () => {
+  let component: TreeComponent;
+  let fixture: ComponentFixture<TreeComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [MatTreeModule, CdkTreeModule],
+      declarations: [TreeComponent],
+      schemas: [CUSTOM_ELEMENTS_SCHEMA]
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(TreeComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should start with a list of defined TreeData', () => {
+    expect(component.dataSource.data).toBeDefined();
+  });
+
+  it('should convert Tree Data to TreeNodes', () => {
+    component.treeData = mockTreeData();
+    const expectedTreeNodes = component.convertToTreeNodes(mockTreeData())
+
+    expect(expectedTreeNodes).toMatchSnapshot();
+  });
+
+  it('should get level of the node', () => {
+    const treeNode = new TreeNode('First test', '54');
+    component.levels.set(treeNode, 0);
+    const expectedLevel = component.getLevel(treeNode);
+
+    expect(expectedLevel).toBe(0);
+  });
+
+  it('should return false if node is NOT expandable', () => {
+    const treeNode = new TreeNode('First test', '54');
+    const expected = component.isExpandable(treeNode);
+
+    expect(treeNode.children.value.length).toBe(0);
+  });
+
+  it('should get children of the node', () => {
+    const childrenNodes = [{ id: '104', nodeName: 'Part paid' }] as TreeNode[];
+
+    const treeNode = new TreeNode('First test', '54', childrenNodes);
+    const expectedChildren = component.getChildren(treeNode);
+
+    expect(expectedChildren).toEqual(treeNode.children);
+  });
+
+  it('should return a node with an incremented level', () => {
+    const childrenNodes = [{ id: '104', nodeName: 'Part paid' }] as TreeNode[];
+
+    const treeNode = new TreeNode('First test', '54', childrenNodes);
+    const expectedNode = component.transformer(treeNode, 1);
+
+    expect(expectedNode).toEqual(treeNode);
+  });
+
+  it('should return true if the node is expandable and has children', () => {
+    const childrenNodes = [{ id: '104', nodeName: 'Part paid' }] as TreeNode[];
+    const treeNode = new TreeNode('First test', '54', childrenNodes);
+
+    expect(treeNode.children.value.length).toBeGreaterThan(0);
+  });
+
+
+  it('should emit the activeNode', () => {
+    const selectedNode = { nodeName: 'First test', id: '95', children: BehaviorSubject, isActive: false };
+    const emittedNode = { key: selectedNode.id, value: selectedNode.nodeName };
+
+    spyOn(component.sendTreeData, 'emit');
+    component.setActiveNode(selectedNode);
+
+    expect(component.sendTreeData.emit).toHaveBeenCalled();
+    expect(component.activeNode).toEqual(emittedNode);
```

Comment:
replace with this `expect(component.sendTreeData.emit).toHaveBeenCalledWith(emittedNode );`